webui/react/src/pages/Authentication.tsx

```diff
@@ -0,0 +1,154 @@
+  const [ badCredentials, setBadCredentials ] = useState(false);
+  const [ canSubmit, setCanSubmit ] = useState(false);
+
+  const queries: Queries = queryString.parse(location.search);
+  const redirectUri = queries.redirect || DEFAULT_REDIRECT;
+
+  const isLogout = location.pathname.endsWith('logout');
+
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+    updateAuth(setAuth, source.token).then(() => setIsLoading(false));
+    return (): void => {
+      source.cancel();
+    };
+  }, [ setAuth ]);
+
+  if (isLogout) {
+    logout({});
+    setAuth({ type: Auth.ActionType.Reset });
+    history.push('/det/login' + location.search);
+  }
+
+  const onFinish = async (creds: FromValues): Promise<void> => {
+    setIsLoading(true);
+    setCanSubmit(false);
+    try {
+      await login(creds as Credentials);
+      updateAuth(setAuth);
+    } catch (e) {
+      setIsLoading(false);
+      setBadCredentials(isLoginFailure(e));
+      const actionMsg = badCredentials ? 'check your username and password.' : 'retry.';
+      handleError({
+        error: e,
+        isUserTriggered: true,
+        message: e.message,
+        publicMessage: `Failed to login. Please ${actionMsg}`,
+        publicSubject: 'Login failed',
+        silent: true,
+        type: badCredentials ? ErrorType.Input : ErrorType.Server,
+      });
+    } finally {
+      setCanSubmit(true);
+    }
+  };
+
+  const onValuesChange = (changes: FromValues, values: FromValues): void => {
+    const hasUsername = !!values.username;
+    setBadCredentials(false);
+    setCanSubmit(hasUsername);
+  };
+
+  const loginForm = (
+    <Form
+      name="login"
+      size="large"
+      onFinish={onFinish}
+      onValuesChange={onValuesChange}
+    >
+      <Form.Item
+        name="username"
+        rules={[
+          {
+            message: 'Please type in your username.',
+            required: true,
+          },
+        ]}
+      >
+        <Input placeholder="Username" prefix={<Icon name="user-small" />} />
+      </Form.Item>
+
+      <Form.Item name="password">
+        <Input.Password placeholder="Password" prefix={<Icon name="lock" />} />
+      </Form.Item>
+
+      {
+        badCredentials &&
+          <p className={[ css['error-message'], css.message ].join(' ')}>
+            Incorrect username or password.
+          </p>
+      }
+
+      <Form.Item>
+        <Button disabled={!canSubmit} htmlType="submit" type="primary">
+          Sign In
+        </Button>
+      </Form.Item>
+    </Form>
+  );
+
+  if (auth.isAuthenticated) {
+    if (isCrossoverRoute(redirectUri)) {
+      crossoverRoute(redirectUri);
+      return <Spinner fullPage />;
+    }
+    return <Redirect to={redirectUri} />;
+  }
+  if (isLogout || isLoading) return <Spinner fullPage />;
+
+  return (
+    <div className={css.base}>
+      <div className={css.content}>
+        {/* DISCUSSION what if we didn't need to add the logo classname and was able to
```

Comment:
remove.
@hkang1 let's chat about this. add it to WebUI arch?

Reply:
added to the doc

Reply:
I understand the `className` motivation now. For the `Logo` component I think of it as an enumeration of variations rather than allowing wrapping components to override what the logo should look like. So using type or some prop to basically pick the look and feel of the logo from a set of presets.

Reply:
hmm even if for layout purposes you might still want to target it :thinking: